### PR TITLE
Draft: Split LJ ixn

### PR DESCRIPTION
**Pull request by jkausrelay**
- Similar to qs, split the LJ parameters
- Needs to be rebased once #1034 is merged

### CODE REVIEW

File: timemachine/potentials/nonbonded.py [hunk 1]
```diff
@@ -539,12 +539,12 @@ def coulomb_interaction_group_energy(q_ligand: Array, q_prefactors: Array) -> fl
...
+    q_ligand: [N_frames, N_lig] array
+    q_prefactors: [N_frames, N_lig] array
 
     Returns
     -------
-    energy: float
+    energy: [N_frames] array
     """
 
     return jnp.dot(q_prefactors, q_ligand)
```
Comment (maxentile) [inline timemachine/potentials/nonbonded.py:550]: docstring was previously correct -- this function applies to a single frame / snapshot.

It can be mapped over multiple frames / snapshots (e.g. using a for loop or vmap or `process_traj_in_chunks`), but for clarity, it is implemented for a single snapshot.

File: timemachine/potentials/nonbonded.py [hunk 2]
```diff
@@ -740,13 +740,13 @@ def lj_interaction_group_energy(sig_ligand, eps_ligand, lj_prefactors):
...
 
     Returns
     -------
-    energy: float
+    energy: [N_frames] array
     """
 
     projection = vmap(basis_expand_lj_atom)(sig_ligand, eps_ligand)
-    return jnp.sum(projection * lj_prefactors)
+    return jnp.sum(jnp.sum(projection * lj_prefactors, axis=2), axis=1)
```
Comment (maxentile) [inline timemachine/potentials/nonbonded.py:752]: Preference: leave this function as defined for a single snapshot, and add a function to apply it to a sequence of snapshots.

(Or update "Assuming lj_prefactors = lj_prefactors_on_snapshot" from docstring to match revised interface.)

File: timemachine/ff/__init__.py [hunk 3]
```diff
@@ -69,6 +73,8 @@ class Forcefield:
         ]
     ]
     lj_handle: Optional[nonbonded.LennardJonesHandler]
+    lj_handle_solv: Optional[nonbonded.LennardJonesSolventHandler]
+    lj_handle_intra: Optional[nonbonded.LennardJonesIntraHandler]
```
Comment (maxentile) [inline timemachine/ff/__init__.py:77]: nit: is there a canonical order for `lj_handle, lj_handle_solv, lj_handle_intra`?

`parameterize_nonbonded` uses a different order `lj_handle, lj_handle_intra, lj_handle_solv`, which seems error-prone.

File: tests/test_jax_nonbonded.py [hunk 4]
```diff
@@ -435,7 +435,7 @@ def f(x, box):
 
     @jit
     def u_batch_test(sig_ligand, eps_ligand, q_ligand):
-        vdw = vmap(lj_interaction_group_energy, (None, None, 0))(sig_ligand, eps_ligand, lj_prefactors)
+        vdw = lj_interaction_group_energy(sig_ligand, eps_ligand, lj_prefactors)
```
Comment (badisa) [inline tests/test_jax_nonbonded.py:438]: Why is this no longer vmap?

**Review by maxentile — APPROVED** [review]

**Review by badisa — APPROVED** [review]

File: tests/test_jax_nonbonded.py (hunk 4 above)
Comment (jkausrelay) [reply tests/test_jax_nonbonded.py]: reverted

File: timemachine/potentials/nonbonded.py (hunk 1 above)
Comment (jkausrelay) [reply timemachine/potentials/nonbonded.py]: reverted

File: timemachine/potentials/nonbonded.py (hunk 2 above)
Comment (jkausrelay) [reply timemachine/potentials/nonbonded.py]: reverted

I see. I was relying on a feature of jnp.dot which made coulomb_interaction_group_energy happen to work on a trajectory (If a is an N-D array and b is a 1-D array, it is a sum product over the last axis of a and b). I'll update the calling code and keep this the same.
